### PR TITLE
feat: enable user input

### PR DESCRIPTION
**Pull request by jacobleesinger**
[Ticket](https://trello.com/c/BsD4nYC0/5870-add-input-format-to-credential-schemas-add-support-for-multi-valued-and-user-inputted-data-fields)

## Summary

Adds `input` property to credential json schemas describing how data should be gotten from the user

## Changes

- added new `values` file
- added `inputTypes` enum
- added options for credentials using select input
- added `input` field to credential json schemas
- added `input` keyword to ajv

## Testing

- tested locally in schema resolver

## Checklist

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas _to a borderline excessive amount_
- [x] I have made any relevant corresponding changes to the documentation including the project readme.
- [x] I have run and tested the changes locally
- [ ] If it is a core feature, I have added an appropriate amount of unit tests.
- [ ] Any dependent changes have been merged and published in upstream projects

### CODE REVIEW

File: src/ajv.ts [hunk 1]
```diff
@@ -94,3 +95,49 @@ const displayFormatDefinition: FuncKeywordDefinition = {
 };
 
 ajv.addKeyword('displayFormat', displayFormatDefinition);
+
+// allows input information describing how data should be collected to be added to the schema
+const inputDefinition: FuncKeywordDefinition = {
```
Comment (rsmets) [inline src/ajv.ts:100]: Was is the input format seemingly recursive? Mind adding a comment explaining?

File: src/values.ts [hunk 2]
```diff
@@ -0,0 +1,1315 @@
+/**
```
Comment (rsmets) [inline src/values.ts:1]: Are these only for use with the input formats or can they be used elsewhere?

**Comment by rsmets** [comment]
Also, looks like you have some conflicts to resolve.

File: src/ajv.ts (hunk 1 above)
Comment (jacobleesinger) [reply src/ajv.ts:100]: It's not recursive, it's just a little complex as the information we need to include is different for each input type

File: src/values.ts (hunk 2 above)
Comment (jacobleesinger) [reply src/values.ts:1]: I guess they could be used anywhere? I thought about having the select input display values be translated via translation layer, but this seemed simpler for now.

Comment (jacobleesinger) [reply src/values.ts:1]: I imagine something like this will eventually become part of the translation layer for data types that are sets of defined values

Comment (jacobleesinger) [reply src/values.ts:1]: something like in this case
```ts
const currencies = [
  {
    iso4217Code: 'AED',
    iso4217CodeAndName: 'AED (UAE Dirham)',
    name: 'UAE Dirham',
    ...
  },
  ...
]
```

**Review by iagormoraes — APPROVED** [review]
Great work, it looks fantastic!

**Pull request merged**